src/py42/clients/securitydata.py

```diff
@@ -322,9 +327,9 @@ def _try_get_security_detection_event_client(self, plan_storage_info):
 
         # otherwise, create it
         if client is None:
-            client = self._storage_client_factory.from_plan_info(
+            client = self._storage_service_factory.create_security_data_service(
                 plan_storage_info.plan_uid, plan_storage_info.destination_guid
-            ).securitydata
+            )
 
             # store this client via its guid so that we don't have to call StorageAuthToken
             # just to determine what storage client to use
```

Comment:
Line 373 needs updating .... can just use `data.get()` and the comment above can be remove or altered

Reply:
@alanag13 This is also another example where we don't need `json.loads()` still